staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -119,36 +120,54 @@ func newDefaultFieldManager(f Manager, typeConverter internal.TypeConverter, obj
 	f = NewLastAppliedManager(f, typeConverter, objectConverter, kind.GroupVersion())
 	f = NewLastAppliedUpdater(f)
 
-	return NewFieldManager(f)
+	return NewFieldManager(f, isSubresource)
 }
 
-// Update is used when the object has already been merged (non-apply
-// use-case), and simply updates the managed fields in the output
-// object.
-func (f *FieldManager) Update(liveObj, newObj runtime.Object, manager string) (object runtime.Object, err error) {
+func decodeManagedFields(liveObj, newObj runtime.Object, isSubresource bool) (Managed, error) {
```

Comment:
This function is only used once and doesn't really add any value, does it?

Reply:
Yeah, I suspect we could do simpler than that somehow

Reply:
I see your point, I kind of like to have in a separate function because in case of an error we can just `return` and the logic is much less nested. Note that with this change we catch an error that was not handled before (see [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go#L149) ).

I have tried to follow your [previous suggestion](https://github.com/kubernetes/kubernetes/pull/91993#discussion_r438301338) but after `isResetManagedFields` has been added I didn't really like how the logic was flowing.

WDYT?

Reply:
I suspect the bug you fixed is the intended behavior and you broke it? Can you describe what you think you've solved? I hope the tests will catch something.

I don't necessarily disagree that the separate function helps, but the content of this function can still be improved.

Reply:
oops you are totally right! I will try to simplify it a bit. I find the `if managed, err = ...` a bit confusing 😬

Reply:
> I hope the tests will catch something.
I think they didn't 😬

Reply:
> This function is only used once and doesn't really add any value, does it?

I think it's complicated enough that the function name is valuable documentation.